File: scripts/layer.lua
-- Copyright (C) 2019  Igara Studio S.A.
--
-- This file is released under the terms of the MIT license.
-- Read LICENSE.txt for more information.

do
  local s = Sprite(32, 64)
  local l = s.layers[1]
  assert(l.parent == s)
  assert(l.opacity == 255)
  assert(l.blendMode == BlendMode.NORMAL)

  l.name = "My Layer"
  l.opacity = 128
  l.blendMode = BlendMode.MULTIPLY
  assert(l.name == "My Layer")
  assert(l.opacity == 128)
  assert(l.blendMode == BlendMode.MULTIPLY)

  l.data = "Data"
  assert(l.data == "Data")
end

File: src/ui/layer.cpp
// Aseprite UI Library
// Copyright (C) 2018-2024  Igara Studio S.A.
// Copyright (C) 2001-2016  David Capello
//
// This file is released under the terms of the MIT license.
// Read LICENSE.txt for more information.

#ifdef HAVE_CONFIG_H
  #include "config.h"
#endif

#include "ui/layer.h"

#include "os/surface.h"

namespace ui {

void UILayer::reset()
{
  m_surface.reset();
}

void UILayer::setSurface(const os::SurfaceRef& newSurface)
{
  m_surface = newSurface;
}

void UILayer::setPosition(const gfx::Point& pos)
{
  m_pos = pos;
}

gfx::Rect UILayer::bounds() const
{
  if (!m_surface)
    return gfx::Rect();

  return gfx::Rect(m_pos.x, m_pos.y, m_surface->width(), m_surface->height());
}

} // namespace ui
